driver/src/bin/main.rs

```diff
@@ -1,21 +1,31 @@
+extern crate graph;
 extern crate simple_logger;
 
+use dfusion_core::database::GraphReader;
+
 use driver::contract::SnappContractImpl;
-use driver::mongo_db::MongoDB;
-use driver::price_finding::LinearOptimisationPriceFinder;
-use driver::price_finding::NaiveSolver;
-use driver::price_finding::PriceFinding;
+use driver::price_finding::{
+    NaiveSolver,
+    LinearOptimisationPriceFinder,
+    PriceFinding
+};
 use driver::run_driver_components;
 
+use graph::log::logger;
+use graph_node_reader::Store as GraphNodeReader;
+
 use std::env;
 use std::thread;
 use std::time::Duration;
 
 fn main() {
+    // driver logger
     simple_logger::init_with_level(log::Level::Info).unwrap();
-    let db_host = env::var("DB_HOST").expect("Specify DB_HOST env variable");
-    let db_port = env::var("DB_PORT").expect("Specify DB_PORT env variable");
-    let db_instance = MongoDB::new(db_host, db_port).unwrap();
+    let graph_logger = logger(false);
+    
+    let postgres_url = env::var("POSTGRES_URL").expect("Specify POSTGRES_URL variable");
+    let store_reader = GraphNodeReader::new(postgres_url, &graph_logger);
+    let db_instance = GraphReader::new(Box::new(store_reader));
     let contract = SnappContractImpl::new().unwrap();
```

Comment:
@bh2smith @fleupold  
Is it right to give the `GraphReader` to the run_driver_components?

It seams to me that the driver need to be more decoupled from the implementation. So it should send just a DbInterface, so we can provide any implementation we want:

![image](https://user-images.githubusercontent.com/2352112/63767095-ee652380-c8cc-11e9-84db-9886cd646554.png)

Reply:
Same would apply to the other params. Shouldn't we pass traits, instead of implementations?

Reply:
Yes it should. I believe this still dates back from our first days in rust and we had trouble getting it to compile when passing traits. But you are totally right. I'll submit a separate PR.